cumulus/client/cirrus-executor/src/bundle_processor.rs

```diff
@@ -512,11 +512,24 @@ where
 
 		for fraud_proof in fraud_proofs {
 			let bad_receipt_number = fraud_proof.parent_number + 1;
-			crate::aux_schema::delete_bad_receipt(
+			// TODO: There is a chance that the fraud proof and the bad receipt are included
+			// in the same block, especially when multiple receipts are included in one block,
+			// which means it's possible we're deleting a receipt which was just inserted above,
+			// so it's more efficient we collect all the write&delete changes before operating
+			// the database.
```

Comment:
Not only that, I suspect it is possible that ER doesn't exist at all

Reply:
It should exist if the executor is honest unless I miss something.

Reply:
Isn't it possible that FP in a block corresponds to ER executor haven't seen yet

Reply:
When the primary block B is imported and processed by the executor, the executor also proceeds to the same height as B, a FP found in primary block B must point to some prior block before B, whose receipt must exist on the executor side at this moment.

Reply:
Okay, thanks for explaining